frontend/src/app/[organisation]/bemanning/api/departments/route.ts

```diff
@@ -7,7 +7,7 @@ export async function GET(request: Request) {
   const organisationName = searchParams.get("organisationName") || "";
 
   const departments =
-    (await fetchWithToken<Department[]>(
+    (await fetchWithToken<Department[], undefined>(
```

Comment:
Why undefined here? Shouldn't there be a response-type etc?

Reply:
This is a part of typescript where I don´t have much knowledge, so I can try and explain what I´ve done and if you have any better ways of solving it i would be happy to hear.

So in fetch/post/put with token i need a returntype and a bodytype, which can change for each call. So they look  like this

```
export async function putWithToken<ReturnType, BodyType>(
  path: string,
  body?: BodyType,
): Promise<ReturnType | undefined> {
  return callApi<ReturnType, BodyType>(path, "PUT", body);
}
```

All of our fetches so far does not take in a body, and our one put call does not need a return. (Hence the never you also commented on). 

So the undefined is not the responsetype, its the body type

Reply:
Good response Sigrid, that's quite correct. 

In generic functions, you can have as many generic types as you want: `myFunc<T, U, ....>`
Most common is to have a generic input and generic output, as we have here. 

However, maybe we should look into making it optional. Or we can just let body be an `Object` type, so we dont need a generic version of that

Reply:
We changed it to not be in fetchCalls 😅